### PR TITLE
detect/transform: from_base64 transform

### PR DESCRIPTION
**Pull request by jlucovsky**
Continuation of #11346 

This PR provides the `from_base64` transform used to match on content decoded with base64 using the mode (default rfc4648).

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: [6487](https://redmine.openinfosecfoundation.org/issues/6487)

Describe changes:
- New transform: `from_base64`
- Documentation
- Keyword parser in rust.

Updates:
- Update fuzzing targets with Rust enum types

### Provide values to any of the below to override the defaults.

To use a pull request use a branch name like `pr/N` where `N` is the
pull request number.

Alternatively, `SV_BRANCH` may also be a link to an
OISF/suricata-verify pull-request.

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1670

### CODE REVIEW

File: rust/src/detect/transform_base64.rs [hunk 1]
```diff
@@ -0,0 +1,441 @@
...
+use nom7::bytes::complete::tag;
+use nom7::character::complete::multispace0;
+use nom7::sequence::preceded;
+use nom7::{Err, IResult};
+use std::str;
+
+#[repr(u8)]
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum DetectBase64Mode {
+    Base64ModeRelax = 0,
```
Comment (inashivb) [inline rust/src/detect/transform_base64.rs:34]: nit: Relax mode does not seem to have been implemented any longer and doesn't do anything. Do you think we should remove it?

File: src/detect-transform-base64.c [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+    return tbd;
+}
+
+/**
+ *  \internal
+ *  \brief Base64 decode the input buffer
+ *  \param det_ctx detection engine ctx
+ *  \param s signature
+ *  \param opts_str transform options, if any
+ *  \retval 0 No decode
```
Comment (inashivb) [inline src/detect-transform-base64.c:85]: nit: did you mean -1 or 0?

File: src/detect-transform-base64.c [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+}
+
+static void TransformFromBase64Decode(InspectionBuffer *buffer, void *options)
+{
+    SCDetectTransformFromBase64Data *b64d = options;
+    const uint8_t *input = buffer->inspect;
+    const uint32_t input_len = buffer->inspect_len;
+    /*
+     * The output buffer is larger than needed. On average,
+     * a base64 encoded string is 75% of the encoded
```
Comment (inashivb) [inline src/detect-transform-base64.c:124]: nit: decoded string is 75%

**Review by inashivb — APPROVED** [review]
Some nits inline but nothing that should block the merge. Tests pretty good. 🚀

**Review by victorjulien — APPROVED** [review]

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 21167

**Comment by victorjulien** [comment]
Merged in #11353, thanks!

File: src/detect-transform-base64.c (hunk 3 above)
Comment (jlucovsky) [reply src/detect-transform-base64.c:124]: Good point ... after the release occurs, I'll update this.

File: rust/src/detect/transform_base64.rs (hunk 1 above)
Comment (jlucovsky) [reply rust/src/detect/transform_base64.rs:34]: Good observation. I'll include that in a followup pr (post release).

**Pull request closed without merging**